### ISSUE TITLE
Dropzone is not accessible to screen readers or keyboard users

### ISSUE THREAD

**Issue by emily-plummer** [okonet/react-dropzone#408]
**Do you want to request a *feature* or report a *bug*?**

- [x] I found a bug
- [ ] I want to propose a feature

**What is the current behavior?**
Navigating to a dropzone element is impossible with a keyboard as `<div>`s are non-focusable by default. Video recording of this happening here: https://cl.ly/0l2A422g3p40 (note that the dotted box is the element using dropzone).

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Open a page using dropzone (make sure that disableClick is not set to true)
2. Tab to the dropzone element using your keyboard (if you're on a mac, you can enable VoiceOver by pressing CMD + F5 which will aid you in navigating the page just using a keyboard)
3. The dropzone element is not in the tab index, and you cannot focus the element

**What is the expected behavior?**
Dropzone should be in the tab index and should handle selections of the element using just a keyboard. I see this a more of a bug than a feature request because clickable elements should be using the semantic `<button>` element which handles this by default.

There are two possible ways to fix this:
1) Change the container element [here](https://github.com/okonet/react-dropzone/blob/master/src/index.js#L337) to be a `<button>` instead of a `<div>` . `<button>` elements are focusable by default and `onClick` handlers are triggered by keyboard selection out of the box :) This is the preferred solution as it's semantically correct. The cons of this solution is that some browsers add default styling to buttons that you'd need to override with css.

2) Add a `tabIndex="0"` to the element. The con of this solution is that you'll need to add handling just for keyboard events. That might look something like this:
``` 
onKeyDown(e) {
    if (e.keyCode !=== '13') return;
    this.onClick(e);
}
...
    return (
      <div
        className={className}
        style={appliedStyle}
        {...divProps/* expand user provided props first so event handlers are never overridden */}
        onClick={this.onClick}
        onDragStart={this.onDragStart}
        onDragEnter={this.onDragEnter}
        onDragOver={this.onDragOver}
        onDragLeave={this.onDragLeave}
        onDrop={this.onDrop}
        onKeyDown={this.onKeyDown}
        ref={this.setRef}
      >
```

**Please mention other relevant information such as the browser version, Operating System and react-dropzone version.**
Browser version: All
Operating System: All
react-dropzone version: 3.12.4
More on accessible buttons here: https://www.paciellogroup.com/blog/2011/04/html5-accessibility-chops-just-use-a-button/

I'd be happy to make a PR for this, but not sure which solution is preferred! Also happy to chat more about accessibility in general and why it's important for this module. Dropzone is a super awesome tool and I'd love to be able to continue using it!

**Comment by okonet** [comment]
Duplicate of https://github.com/okonet/react-dropzone/issues/161 (see the discussion there) + https://github.com/okonet/react-dropzone/pull/336

Can you collaborate on them please?

**Comment by xiaochenzhang** [comment]
Hi @okonet , 

sorry to comment on a "closed" issue. But I still cannot get focus on the dropzone when using keyboard tab (on MacOS 11.4, Chrome Version 92.*). I saw this issue is fixed in #161 , but I don't know what I did wrong or missed in my implementation, could you guide me a bit? Here is my code (and lib version):

```
    "react": "16.10.0",
    "react-dropzone": "^10.2.1",

      <Dropzone onDrop={uploadFile} accept=".pdf, .png, .jpeg, .jpg, .doc, .docx">
        {({ getRootProps, getInputProps }) => (
          <section className={classes.dropzoneContainer}>
            <div {...getRootProps()} className={classes.dropzoneInner}>
              <div className={classes.dropSection}>
                <CloudUploadIcon className={classes.uploadIcon} />
                <input {...getInputProps()} />
                <p className={classes.documentName}>Upload</p>
              </div>
            </div>
          </section>
        )}
      </Dropzone>

```

**Comment by rolandjitsu** [comment]
> Hi @okonet ,
> 
> sorry to comment on a "closed" issue. But I still cannot get focus on the dropzone when using keyboard tab (on MacOS 11.4, Chrome Version 92.*). I saw this issue is fixed in #161 , but I don't know what I did wrong or missed in my implementation, could you guide me a bit? Here is my code (and lib version):
> 
> ```
>     "react": "16.10.0",
>     "react-dropzone": "^10.2.1",
> 
>       <Dropzone onDrop={uploadFile} accept=".pdf, .png, .jpeg, .jpg, .doc, .docx">
>         {({ getRootProps, getInputProps }) => (
>           <section className={classes.dropzoneContainer}>
>             <div {...getRootProps()} className={classes.dropzoneInner}>
>               <div className={classes.dropSection}>
>                 <CloudUploadIcon className={classes.uploadIcon} />
>                 <input {...getInputProps()} />
>                 <p className={classes.documentName}>Upload</p>
>               </div>
>             </div>
>           </section>
>         )}
>       </Dropzone>
> ```

Could you open a new issue and illustrate what you're experiencing? Also, what's your OS, browser, etc? I cannot reproduce on the main docs website.
